nodelet_topic_tools/include/nodelet_topic_tools/nodelet_lazy.h

```diff
@@ -0,0 +1,290 @@
+      ros::WallDuration(5),
+      &NodeletLazy::warnNeverSubscribedCallback,
+      this,
+      /*oneshot=*/true);
+  }
+
+  /** @brief
+    * Post processing of initialization of nodelet.
+    * You need to call this method in order to use always_subscribe
+    * feature.
+    */
+  virtual void onInitPostProcess()
+  {
+    if (!lazy_)
+    {
+      boost::mutex::scoped_lock lock(connection_mutex_);
+      ever_subscribed_ = true;
+      subscribe();
+    }
+  }
+
+  /** @brief
+    * callback function which is called when new subscriber come
+    */
+  virtual void connectionCallback(const ros::SingleSubscriberPublisher& pub)
+  {
+    if (verbose_connection_)
+    {
+      NODELET_INFO("[%s] [nodelet_topic_tools::NodeletLazy::connectionCallback] New connection or disconnection is detected",
+                   nodelet::Nodelet::getName().c_str());
+    }
+    if (lazy_)
+    {
+      boost::mutex::scoped_lock lock(connection_mutex_);
+      for (size_t i = 0; i < publishers_.size(); i++)
+      {
+        ros::Publisher pub = publishers_[i];
+        if (pub.getNumSubscribers() > 0)
+        {
+          if (!ever_subscribed_)
+          {
+            ever_subscribed_ = true;
+          }
+          if (connection_status_ != SUBSCRIBED)
+          {
+            if (verbose_connection_)
+            {
+              NODELET_INFO("[%s] [nodelet_topic_tools::NodeletLazy::connectionCallback] Subscribe input topics",
+                           nodelet::Nodelet::getName().c_str());
+            }
+            subscribe();
+            connection_status_ = SUBSCRIBED;
+          }
+          return;
+        }
+      }
+      if (connection_status_ == SUBSCRIBED)
+      {
+        if (verbose_connection_)
+        {
+          NODELET_INFO("[%s] [nodelet_topic_tools::NodeletLazy::connectionCallback] Unsubscribe input topics",
+                       nodelet::Nodelet::getName().c_str());
+        }
+        unsubscribe();
+        connection_status_ = NOT_SUBSCRIBED;
+      }
+    }
+  }
+
+  /** @brief
+    * callback function which is called when walltimer
+    * duration run out.
+    */
+  virtual void warnNeverSubscribedCallback(const ros::WallTimerEvent& event)
+  {
+    if (!ever_subscribed_)
+    {
+      NODELET_WARN("[%s] [nodelet_topic_tools::NodeletLazy::warnNeverSubscribedCallback] this node/nodelet subscribes topics only when subscribed.",
+                   nodelet::Nodelet::getName().c_str());
+    }
+  }
+
+
+  /** @brief
+    * This method is called when publisher is subscribed by other
+    * nodes.
+    * Set up subscribers in this method.
+    */
+  virtual void subscribe() = 0;
+
+  /** @brief
+    * This method is called when publisher is unsubscribed by other
+    * nodes.
+    * Shut down subscribers in this method.
+    */
+  virtual void unsubscribe() = 0;
+
+  /** @brief
+    * Advertise a topic and watch the publisher. Publishers which are
+    * created by this method.
```

Comment:
Rephrase this as a single sentence? the meaning of "Publishers which are created by this method" is unclear to me. Did you mean "update the list of Publishers created by this method" ?